src/matchPath.js

```diff
@@ -47,7 +47,7 @@ function matchPathBase(end, routePath, urlPath) {
   const params = Object.create(null);
   const path = m[0];
 
-  for (let i = 1; i < m.length; i++) {
+  for (let i = 1; i < m.length; i += 1) {
```

Comment:
Well, Airbnb styleguide prescribes avoiding for-loops altogether:
https://github.com/airbnb/javascript#iterators--nope

Using addition assignment here looks strange to say the least.